firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -221,8 +226,31 @@ public void onFailure(@NonNull Exception e) {
    *     is cached from checkForUpdate
    */
   @NonNull
-  public UpdateTask updateApp() {
-    return (UpdateTask) Tasks.forResult(new UpdateState(0, 0, UpdateStatus.PENDING));
+  public UpdateTask updateApp() throws FirebaseAppDistributionException {
+
+    if (updateAppTaskCompletionSource != null
+        && !updateAppTaskCompletionSource.getTask().isComplete()) {
+      updateAppCancellationSource.cancel();
+    }
+
+    updateAppCancellationSource = new CancellationTokenSource();
+    updateAppTaskCompletionSource =
+        new TaskCompletionSource<>(updateAppCancellationSource.getToken());
+    Context context = firebaseApp.getApplicationContext();
+
+    if (appDistributionReleaseInternal == null) {
+      throw new FirebaseAppDistributionException(
+          context.getString(R.string.no_update_available),
+          FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE);
+    }
+
+    if (appDistributionReleaseInternal.getBinaryType() == BinaryType.AAB) {
+      redirectToPlayForAabUpdate(appDistributionReleaseInternal.getDownloadUrl());
+    } else {
+      throw new UnsupportedOperationException("Not yet implemented.");
+    }
+
```

Comment:
I might have confused myself a little on the UpdateTask method and how that's going to work for APKs. lets sync on this at some point.

Reply:
Also sidenote, a customer could call `addProgressListener` on the UpdateTask and they should get `UpdateStatus.REDIRECTED_TO_PLAY` as the response. This might require some rework to store the UpdateTaskImpl as a class variable instead of `updateAppTaskCompletionSource` and call the updateProgress method